### PR TITLE
MessageListAdapter refactoring

### PR DESCRIPTION
**Pull request by cketti**
Cleaned up version of #4109

### CODE REVIEW

**Comment by bhm** [comment]
This branch is few commits behind. Git failed to push on the subway. Rebase with the feature branch should do it.

**Comment by cketti** [comment]
It's up to date with the current state of PR #4109. The number of commits is smaller because I combined several of them to clean up the history.

File: app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+        previewTextColor = array.getColor(3, Color.BLACK)
+        activeItemBackgroundColor = array.getColor(4, Color.BLACK)
+        selectedItemBackgroundColor = array.getColor(5, Color.BLACK)
+        readItemBackgroundColor = array.getColor(6, Color.BLACK)
+        unreadItemBackgroundColor = array.getColor(7, Color.BLACK)
+
+        array.recycle()
+    }
+
+    private inline val previewLines: Int
```
Comment (bhm) [inline app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: Missing the MessageListAppearance data class.

Comment (cketti) [reply app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: Looks like this change didn't make it into #4109. I suggest we add the class in a separate pull request after this one has been merged.

Comment (bhm) [reply app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: Sure.

**Pull request merged**